lang/py/src/avro/io.py

```diff
@@ -580,16 +696,36 @@ def read_data(self, writers_schema, readers_schema, decoder):
     elif writers_schema.type == 'string':
       return decoder.read_utf8()
     elif writers_schema.type == 'int':
-      return decoder.read_int()
+      if hasattr(writers_schema, 'logical_type'):
+        if writers_schema.logical_type == constants.DATE:
+          return decoder.read_date_from_int()
+        elif writers_schema.logical_type == constants.TIME_MILLIS:
+          return decoder.read_time_millis_from_int()
```

Comment:
@premsantosh i would lean on dropping this issue because such a situation where .avro file contains a time type (int or long), contained a neg value will never occur. If it did then some implementation of avro decoder must have been buggy.